core/primitives/src/account.rs

```diff
@@ -159,91 +199,73 @@ impl BorshSerialize for Account {
                 storage_usage: self.storage_usage,
             }
             .serialize(writer),
+            _ => SerializableAccount {
```

Comment:
```suggestion
            AccountVersion::V2 => SerializableAccount {
```

It's better to default to exhaustive matches: https://youtu.be/-J8YyfrSwTk?t=1819

Reply:
This logic would be the same for all future versions, so this is by design